minicluster/src/main/java/org/apache/accumulo/minicluster/MiniAccumuloConfig.java

```diff
@@ -108,6 +108,19 @@ public MiniAccumuloConfig setZooKeeperStartupTime(long zooKeeperStartupTime) {
   }
 
   /**
+   * Configure an existing ZooKeeper instance to use. Calling this method is optional. If not set, a new ZooKeeper instance is created.
+   *
+   * @param existingZooKeepers
+   *          Connection string for a already-running ZooKeeper instance
+   *
+   * @since 1.7.0
+   */
+  public MiniAccumuloConfig setExistingZooKeepers(String existingZooKeepers) {
```

Comment:
> MiniAccumuloConfig is leaked via a public constructor and the getConfig method on MiniAccumuloCluster

Yup, that's by design.

> If MiniAccumuloConfig doesn't have a setter for the existing zookeepers, then it can't be used by an integration test or in general.

Right, it would need to be used via `MiniAccumuloConfigImpl` and `MiniAccumuloClusterImpl`. We already have many methods hidden on the impl that are not exposed in via the non-impl variants. Nearly all of our ITs in Accumulo use the impl variants already IIRC.